在 Python 2.5 中， with 关键字被加入。它将常用的 try ... except ... finally ... 模式很方便的被复用。看一个最经典的例子：

with open('file.txt') as f:
  content = f.read()

在这段代码中，无论 with 中的代码块在执行的过程中发生任何情况，文件最终都会被关闭。
如果代码块在执行的过程中发生了一个异常，那么在这个异常被抛出前，程序会先将被打开的文件关闭。